torch/cuda/__init__.py

```diff
@@ -211,10 +211,14 @@ def _check_bf16_tensor_supported(device: _device_t):
 
 def is_tf32_supported() -> bool:
     r"""Return a bool indicating if the current CUDA/ROCm device supports dtype tf32."""
-    # Check for ROCm.  If true, return false, since PyTorch does not currently support
-    # tf32 on ROCm.
+    # For AMD GPUs, tf32 is supported on mi300.
     if torch.version.hip:
-        return False
+        prop = torch.cuda.get_device_properties(torch.cuda.current_device())
+        MI300_ARCH = ("gfx942",)
+        if prop.gcnArchName.split(":")[0] in MI300_ARCH:
+            return True
+        else:
+            return False
```

Comment:
Don't need to split gcnArchName.  It's all a str, so just look for your gfx target.  Also, we've been using "gfx94" elsewhere.  Lastly, `get_device_properties()` without an explicit device ordinal will internally get the current device ordinal, so we can shorten that call.  In the future as our tf32 support expands we will update this logic.
```suggestion
        prop = torch.cuda.get_device_properties()
        return "gfx94" in prop.gcnArchName
```